scripts/e2e/e2e.sh

```diff
@@ -184,6 +184,14 @@ go get -u github.com/vmware/govmomi/govc
 # Push new container images
 # TODO the `-k` flag here is a workaround until we can set GCR_KEY_FILE properly
 hack/release.sh -t pr -p -k /root/.capv/keyfile.json
+
+# export variables needed to generate spec files
+export_base64_value "VSPHERE_SERVER" "${GOVC_URL}"
+export_base64_value "VSPHERE_USERNAME" "${GOVC_USERNAME}"
+export_base64_value "VSPHERE_PASSWORD" "${GOVC_PASSWORD}"
+vsphere_controller_version="gcr.io/cluster-api-provider-vsphere/pr/manager:${VERSION}"
+export_base64_value "VSPHERE_CONTROLLER_VERSION" "${vsphere_controller_version}"
```

Comment:
Hi @ykakarap,

I was just playing with these changes a bit to understand why things were failing... The deletion step aside (looking at the logs, we're not actually getting that far -- the pod for the test never comes up), I realized there is a problem here. None of these items should be exported as base64. These are being used directly in your `generate-e2e-spec.sh` script, and thus the the base64 encoded value is ending up in things like `cluster.yaml` when they should be plain text.

The only items that should be base64-encoded are those that end up populating the secret, found at `bootstrap_secret.template`, so it's not something you should have to worry about at all. A quick aside, even *those* items don't need to be base64, so long as the Secrete is defined as `StringData` instead of `OpaqueData`, which is a recent change we made to our docs. That would be a great simplification of this test, but is definitely out of scope for this PR...

Instead I think you should `export` these values directly, so they can be passed to your `generate_e2e_spec.sh` script.

The reason this worked before, is because in the old tempates (https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/a0f94dfd9dc5a479cbb9381391d3aa9bfdb81a24/scripts/e2e/bootstrap_job/spec/cluster.template#L16-L18), we would populate from env vars that are part of the boostrap job. Basically, [this step](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/a0f94dfd9dc5a479cbb9381391d3aa9bfdb81a24/scripts/e2e/bootstrap_job/clusterctl.sh#L45-L47) filled in the templates with values from [here](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/a0f94dfd9dc5a479cbb9381391d3aa9bfdb81a24/scripts/e2e/bootstrap_job.template#L43-L57). In other words, those env vars are already set and present in the environment when that job is running.

So we can actually do one of two things here: generate files that still have things like `${VSPHERE_SERVER}` in them *literally*, like we did before, and still do some sort of env substitution, or, since when running on Prow we already know what those values are going to be, we could just set them directly. I know Andrew was thinking the former, which is why he gave examples of how to use the `manifests` docker image to create YAML files that held a place holder for the env var, rather than expecting to use an actual env var, so that's probably the better choice.

If we take that route, none of these exports should be necessary at all. You want to use hardcoded values that produce placeholders from the `manifests` image, and then still do an env var substitution in `clusterctl.sh`.

I know that's a bit strange, but hopefully that helps. :)

Reply:
I originally implemented (using the `generate-yaml.sh`) the logic as suggested by Andrew in the issue description where we use the literally strings like '$VSPHERE_USERNAME' in spec file. Andrew [commented](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/457#discussion_r307013468) on that implementation that this will not be needed anymore when we switch to the docker `manifests`. 

I realize both approaches are possible at this moment. So, do we have a preference for one approach over the other? Does one approach provide any advantage over the other?

Reply:
Hmm, interesting. Yeah, I don't know if we actually *need* to use the secret for that info... We have it when we generate the manifests when running on Prow. If we go that direction instead, you still won't want to use the `base64` exports -- you'll just want to do a straight `export` instead so that it can get picked up in the `docker run` command.

Reply:
As of now, I am planning to use the `docker run` command to generate the manifests with the actual values in place.

Point noted about `base64` exporting. Thanks about that :)